parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReaders.java

```diff
@@ -1132,6 +1144,46 @@ private TripleIterator<?> firstNonNullColumn(List<TripleIterator<?>> columns) {
     }
   }
 
+  private static class StructLikeReader<T extends StructLike> extends StructReader<T, T> {
+    private final Types.StructType struct;
+    private final DynConstructors.Ctor<T> ctor;
+
+    StructLikeReader(
+        List<ParquetValueReader<?>> readers, Types.StructType struct, Class<T> structLikeClass) {
+      super(readers);
+      this.struct = struct;
+      this.ctor =
+          DynConstructors.builder(StructLike.class)
+              .hiddenImpl(structLikeClass, Types.StructType.class)
+              .hiddenImpl(structLikeClass)
```

Comment:
should default/empty constructor be used here? I though we would require a struct type.

Reply:
I'm mimicking the Avro code here although we can't actually have empty structs here here ...

https://github.com/apache/iceberg/blob/7eef46da0d00edb79b20d24cb7806f6ea8da4000/core/src/main/java/org/apache/iceberg/avro/InternalReaders.java#L55-L60